objects/src/accounts/storage/mod.rs

```diff
@@ -283,6 +279,18 @@ impl AccountStorage {
             storage_map.apply_delta(map_delta)?;
         }
 
+        // --- update storage slots -------------------------------------------
+        // this will also update roots of updated storage maps, and thus should be run after we
+        // update storage maps - otherwise the roots won't match
```

Comment:
I think the order doesn't matter, the storage root update is queued in `updated_items` and the maps update in `updated_maps`. Both are triggered in MASM in `set_map_item`

Reply:
In combination with the code I lined above, the order does matter (otherwise tests won't pass). But also, this code is separate from MASM: when executing MASM we are building the delta, here we are applying it - which happens later.